### PR TITLE
DAOS-7821 ga: Improve github actions coverage and resiliance.

### PR DESCRIPTION
**Pull request by ashleypittman**
Use ubuntu 21.04 in GA for testing as it's newer.
Remove psm2 and psm3 from the build.
Seperate codespell into it's own workflow.
Update the newer process to build Java in docker.
Remove the CentOS 8 and Ubuntu 20.04 builds from older
process as it's now replicated in the newer one.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: site_scons/components/__init__.py [hunk 1]
```diff
@@ -132,20 +111,12 @@ def define_mercury(reqs):
...
                           '--without-gdrcopy ' +
-                          OFI_DEBUG +
-                          exclude(reqs, 'psm2',
-                                  '--enable-psm2' +
-                                  check(reqs, 'psm2',
-                                        "=$PSM2_PREFIX "
-                                        'LDFLAGS="-Wl,--enable-new-dtags ' +
-                                        '-Wl,-rpath=$PSM2_PREFIX/lib64" ',
-                                        ''),
-                                  ''),
```
Comment (rogue-developer) [inline site_scons/components/__init__.py:144]: Why don't we explicitly disable it? Just like the other providers
`--disable-psm2`

File: site_scons/components/__init__.py [hunk 2]
```diff
@@ -108,20 +101,6 @@ def define_mercury(reqs):
...
-                          '       -e "s/\\(INSTALL_LIB_TARG=' +
-                          '\\/usr\\/lib\\)64/\\1/" ' +
-                          '       -e "s/\\(INSTALL_LIB_TARG=\\)\\/usr/\\1/" ' +
-                          'Makefile compat/Makefile',
-                          'make $JOBS_OPT LIBDIR="/lib64"',
-                          'make DESTDIR=$PSM2_PREFIX LIBDIR="/lib64" install'],
-                headers=['psm2.h'],
-                libs=['psm2'])
-
     if reqs.target_type == 'debug':
```
Comment (frostedcmos) [inline site_scons/components/__init__.py:104]: I dont think we can remove psm2 support yet. @johannlombardi ?

File: site_scons/components/__init__.py [hunk 3]
```diff
@@ -132,20 +111,12 @@ def define_mercury(reqs):
                 commands=['./autogen.sh',
                           './configure --prefix=$OFI_PREFIX ' +
                           '--disable-efa ' +
+                          '--disable-psm3 ' +
                           '--without-gdrcopy ' +
```
Comment (frostedcmos) [inline site_scons/components/__init__.py:115]: if we decide to remove psm2 support then for now (until we update to most recent OFI release) we need to explicitly disable it via -disable-psm2. most providers in OFI today are compiled as 'auto', where they auto-detect if they can be compiled, and if so, they get enabled.  
The change that landed recnetly to OFI that we requested adds new --enable-only option to OFI which then disables all providers by default except the ones we explicitly enable via --enable-[provider] flag

File: site_scons/components/__init__.py (hunk 1 above)
Comment (rogue-developer) [reply site_scons/components/__init__.py:144]: If it is decided to remove psm2 support. You may consider removing psm2 from the scons building EXCLUDE list
https://github.com/daos-stack/daos/blob/0e8cb15cd627441f8e16cf12c267869cb934d3fb/site_scons/prereq_tools/base.py#L683

Comment (rogue-developer) [reply site_scons/components/__init__.py:144]: If it is decided to remove psm2 support. You may consider removing psm2 from the scons building EXCLUDE list
https://github.com/daos-stack/daos/blob/0e8cb15cd627441f8e16cf12c267869cb934d3fb/site_scons/prereq_tools/base.py#L683

Comment (ashleypittman) [reply site_scons/components/__init__.py:144]: Yes, agreed.  I had some comments with Jeff about this before he went out on holiday.

Given the way this is going I'm leaning back more towards having a PR just to strip out the PSM2 code.  The reason the change to disable it is in here is that it doesn't build on the newer Ubuntu versions.

File: src/cart/crt_hg.c [hunk 4]
```diff
@@ -368,21 +368,9 @@ crt_provider_ip_str_get(int provider)
...
 }
 
-static bool
-crt_provider_is_block_mode(int provider)
-{
-	if (provider == CRT_NA_OFI_PSM2)
-		return false;
-
-	return true;
-}
```
Comment (frostedcmos) [inline src/cart/crt_hg.c]: we want to keep this for now for new provider which is likely non blocking

File: src/cart/crt_hg.c [hunk 5]
```diff
@@ -574,10 +562,7 @@ crt_hg_class_init(int provider, int idx, hg_class_t **ret_hg_class)
 	if (rc != 0)
 		D_GOTO(out, rc);
 
-	if (crt_provider_is_block_mode(provider))
-		init_info.na_init_info.progress_mode = 0;
-	else
-		init_info.na_init_info.progress_mode = NA_NO_BLOCK;
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/cart/crt_hg.c]: I backed out this part of the change already.

File: site_scons/components/__init__.py (hunk 1 above)
Comment (rogue-developer) [reply site_scons/components/__init__.py]: I created #5927 to address [DAOS-7849](https://jira.hpdd.intel.com/browse/DAOS-7849)

**Review by brianjmurrell — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**